pkg/logging/http.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+
+	"net/http"
+	"time"
+	"github.com/thanos-io/thanos/pkg/server/http/middleware"
+)
+
+type LogServerMiddleware struct {
+	opts   *options
+	logger log.Logger
+}
+
+// Things for logger
+// * request-id
+// * err code mapping to level
+// * duration of request
+
+func (c *LogServerMiddleware) HTTPMiddleware(name string, next http.Handler) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		// If request logging is disabled
+		if !c.opts.shouldLog(name) {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		logger := c.logger
```

Comment:
Do we need this statement? Why not doing `c.logger.With...` below?

Reply:
Changed